### ISSUE TITLE
ArrayIndexOutOfBoundsException still not fixed

### ISSUE THREAD

**Issue by MCEventHorizon** [Electroblob77/Wizardry#238]
Minecraft version: 1.12.2
Wizardry version: 4.2.2
Environment: Singleplayer/Server

Issue details: Issue that's kicking players off the server when casting spells or attacking mobs still not fixed.

Link to crash report (if applicable): https://gist.github.com/MCEventHorizon/d1eba835ac26256cffacc618da2f8f28

**Comment by Electroblob77** [comment]
Make sure that all players have the latest update and have deleted their old options.txt, just in case. I will see what I can do, but it has to be some kind of compatibility issue as it now works correctly for me.

**Comment by SirAron111** [comment]
Yes can confirm this. The error still happens it happened to me in the new version 4.2.2 when i accessed the sound volume options. I did let a completely new options.txt generate and used that but it happened the second time I tried to change volumes.
https://paste.dimdev.org/wojewadize.mccrash

Also still happens on skills like flame ray https://paste.dimdev.org/alojubuqiv.mccrash

Maybe you are using some kind of way to set the configs and sound access that was meant for 1.13 and up like you did with the data packs?
Or you could try having the sound value of the spells just be adjusted via your config not the options menu and see if that fixes it if you don’t access the standard sound system at all.

**Comment by ghost** [comment]
I'm also still having the same issue, as are other users of the same modpack. Deleting options.txt fixes the problem once, but it returns the second time you run the game.

https://paste.dimdev.org/useheragaq.mccrash

**Comment by SirAron111** [comment]
And deleting the options.txt isn’t really a fix at all since you would need to readjust ALL your settings keybinds etc which is hell in larger packs. So the question should be why does deleting the options temporarily remove the crash there must be an issue with how ebwiz reads and writes the options or interacts with the process most other mods seem to use to do so.

**Comment by Electroblob77** [comment]
As detailed in the above commit, this issue has been temporarily fixed by disabling the spells sound category. The feature will be reinstated once a proper fix is found. Please don't close this issue in the meantime.
